app/models/event.rb

```diff
@@ -66,11 +66,7 @@ def suggestion_summary
   end
 
   def keynote_speakers
-    talks.select { |talk|
-      talk.title.start_with?("Keynote: ") ||
-        talk.title.include?("Opening Keynote") ||
-        talk.title.include?("Closing Keynote")
-    }.flat_map(&:speakers)
+    speakers.merge(talks.keynote)
```

Comment:
I'm surprised this works 😅 

Does it know how to merge speakers with talks and its speakers?

Reply:
It works because it's not merge from Hash or Array, it's [merge from ActiveRecord](https://apidock.com/rails/ActiveRecord/SpawnMethods/merge) ☺️.

I did that to re-use the scope provided by the enum. It's the same as doing this:

```ruby
speakers # this is the has many through association in event
  .where(talk: { kind: 'keynote' }
```

Reply:
Ah I see! I think I never used ActiveRecord's merge with two different models/table, thanks for the context

Reply:
Wondering if we should promote it to a proper association, something like:

```ruby
 has_many :keynotes, -> { keynote }, class_name: "Talk"
 has_many :keynote_speakers, through: :keynotes, source: :speakers
```

Reply:
@marcoroth I noticed you marked this as resolved. Do you still think it should be promoted to an association?

I'm fine with keeping it as a method, but I'm open to making it an association if you think it's better.

Reply:
I was just thinking out loud, but didn't want to keep requesting things. We can refactor it later if we feel like it needs to be an association 😊